src/expressions/functions/builtInFunctions.node.ts

```diff
@@ -112,6 +112,84 @@ const fnHasChildren: FunctionDefinitionType = function(
 	});
 };
 
+const fnPath: FunctionDefinitionType = function(
+	_dynamicContext,
+	executionParameters,
+	_staticContext,
+	nodeSequence: ISequence
+) {
+	return zipSingleton([nodeSequence], ([nodeValue]: (Value | null)[]) => {
+		const node: ConcreteNode = nodeValue ? nodeValue.value : null;
+
+		if (!node) {
+			return sequenceFactory.empty();
+		}
+
+		let result = '';
+
+		function getChildIndex(child: ConcreteChildNode): number {
+			let i = 0;
+			const childNodeType = child.nodeType;
+			const childNodeName = child.nodeName;
+			while (child !== null) {
+				if (childNodeType === child.nodeType && childNodeName === child.nodeName) {
```

Comment:
I'm missing tests for duplicate namespace URIs. I hoped the QT3 tesst offered them, but they don't...

Could you add the following tests:

```
let $dom := <xml>
  <a:a xmlns:a="http://test"/>
  <a:a xmlns:a="http://test2"/>
</xml>
return path(/xml/*:a[2])

(: Should output "fn:root()/Q{http://test2}a[1]" :)
```

```
let $dom := <xml>
  <a:a xmlns:a="http://test"/>
  <b:a xmlns:b="http://test"/>
</xml>
return path(/xml/*:a[2])

(: Should output "fn:root()/Q{http://test}a[2]" :)
```

Reply:
done